### PR TITLE
Feature/mark as owned

### LINKED ISSUE #3: #3 Mark Plants as Owned
## Value proposition
As a user
I want to mark the plants I own,
so that I can easily track and manage the care of my personal collection.
## Description
Image

## Acceptance criteria

- [ ] Each listed plant includes a “Mark as Owned” option.
- [ ] Clicking the “Mark as Owned” button of an unmarked plant adds it to the “My Plants” list.
- [ ] Clicking the same button again removes the plant from the “My Plants” list.
- [ ] The “My Plants” list is displayed on a separate page dedicated to owned plants.
- [ ] There is a clearly visible link on the homepage to navigate to the “My Plants” page.
- [ ] There is a clearly visible link on the “My Plants” page to navigate back to the homepage.
- [ ] Owned plants are visually highlighted by a distinct icon or style to differentiate them from unowned plants.
- [ ] If there are no owned plants in the “My Plants” list, a message is displayed to the user indicating that no plants have been marked as owned.

## Tasks

- [ ]  Create feature branch feature/mark-as-owned
- [ ]  create mark owned button in PlantCard
- [ ] create function to toggle mark owned button
- [ ] create page myPlants
- [ ] create a Footer component with a Link to Homepage and a Link to My plants

### PR DESCRIPTION
**Pull request by ana-nova**

### CODE REVIEW

File: components/Footer.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import HouseLine from "@/public/Icons/home-4-line.svg";
+import styled from "styled-components";
+import PlantFavIcon from "@/public/Icons/plant-line.svg";
+
+export default function Footer() {
+  return (
+    <StyledFooter>
+      <StyledLink href={"/"}>
```
Comment (JessicaLoers) [inline components/Footer.js]: Consider adding `aria-label` attributes here for accessibility, so screen readers can interpret the purpose of each link.  Because screen readers might not interpret the purpose of an icon-only link without additional context, so here’s how you could improve it:

```suggestion
      <StyledLink href={"/"} aria-label="Go to homepage">
```

File: components/Footer.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+        <HouseLine />
+      </StyledLink>
+      <StyledLink href={"/myPlants"}>
+        <PlantFavIcon />
```
Comment (JessicaLoers) [inline components/Footer.js]:
```suggestion
      <StyledLink href={"/myPlants"} aria-label="Go to my plants">
```

File: components/Footer.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+      </StyledLink>
+    </StyledFooter>
+  );
+}
+
+const StyledFooter = styled.div`
```
Comment (JessicaLoers) [inline components/Footer.js]: Since this is a footer, please use the `<footer>` HTML tag instead of a `<div>`. This semantic change will make the structure more accessible and improve SEO.
```suggestion
const StyledFooter = styled.footer
```

File: components/Footer.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+  position: fixed;
+  bottom: 0;
+  width: 100%;
+  background-color: lightgray;
+  padding: 10px;
+  display: flex;
+  justify-content: space-around;
+`;
+
+const StyledLink = styled.a`
```
Comment (JessicaLoers) [inline components/Footer.js]: Instead of `<a>`, please use the Next.js’s [Link component](https://nextjs.org/docs/pages/api-reference/components/link)  for client-side routing. 
 You can then nest the icon inside the Link.
 
 ```suggestion
const StyledLink = styled.(Link)`
```

File: components/PlantCard.js [hunk 5]
```diff
@@ -1,15 +1,17 @@
 import styled from "styled-components";
+import PlantFavIcon from "@/public/Icons/plant-line.svg";
+import PlantFavIconFill from "@/public/Icons/plant-fill.svg";
 
-export default function PlantCard({ plant }) {
+export default function PlantCard({ plant, toggleFavourite }) {
   return (
     <Container>
+      <button onClick={() => toggleFavourite(plant.id)}>
```
Comment (JessicaLoers) [inline components/PlantCard.js]: Consider adding an aria-label for the button to make it accessible. This will help screen readers understand its purpose, especially if the icon changes based on state.

```suggestion
      <button aria-label={plant.isFavourite ? "Remove from favorites" : "Add to favorites"} onClick={() => toggleFavourite(plant.id)} >
```

File: components/PlantCard.js [hunk 6]
```diff
@@ -1,15 +1,17 @@
...
       <h2>{plant.name}</h2>
       <StyledParagraph>{plant.botanicalName}</StyledParagraph>
 
-      <RoundImage
-        alt={`Image of ${plant.name}`}
-        src={plant.imageUrl}
-      ></RoundImage>
+      <RoundImage alt={`Image of ${plant.name}`} src={plant.imageUrl} />
 
       <StyledLink href={`/${plant.id}`}>see more</StyledLink>
```
Comment (JessicaLoers) [inline components/PlantCard.js:28]: Not part of the PR, but please Next.js's `Link` component instead of `<a>`. This allows for faster client-side navigation and integrates with Next.js routing.

File: components/PlantList.js [hunk 7]
```diff
@@ -1,17 +1,16 @@
-import { plants } from "@/assets/plants";
 import styled from "styled-components";
 import PlantCard from "./PlantCard";
 
-export default function PlantList() {
+export default function PlantList({ plants, toggleFavourite }) {
   if (!plants || plants.length === 0) return <p>No plants found</p>;
 
   return (
     <div>
```
Comment (JessicaLoers) [inline components/PlantList.js]: Why you’re using a `<div>` as the parent element here.  For semantic HTML, it’s often better to use a `<main>` element as the main wrapper for primary content by indicating that this section is the central content of the page.

Option: If this header applies specifically to the plant collection, you could place it within `<main>`.  (replace the `div` with a `main` tag)
This would signify that the header relates directly to the content in this section.

File: pages/index.js [hunk 8]
```diff
@@ -1,5 +1,26 @@
...
+        plant.id === id ? { ...plant, isFavourite: !plant.isFavourite } : plant
+      )
+    );
+  }
+
+  return (
+    <div>
+      <PlantList plants={plants} toggleFavourite={toggleFavourite} />
+      <Footer />
+    </div>
```
Comment (JessicaLoers) [inline pages/index.js]: - As mentioned above.  Consider using a `<main>` element instead of `<div>` to wrap the primary content of the page.

- Also, if you plan to reuse the layout with a header or other components, consider creating a Layout component to wrap content and ensure consistent structure across pages.
```suggestion
    <main>
      <PlantList plants={plants} toggleFavourite={toggleFavourite} />
      <Footer />
    </main>
```

File: pages/myPlants.js [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+    ? plants.filter((plant) => plant.isFavourite)
+    : [];
+
+  function toggleFavourite(id) {
+    setPlants((prevPlants) =>
+      prevPlants.map((plant) =>
+        plant.id === id ? { ...plant, isFavourite: !plant.isFavourite } : plant
+      )
+    );
+  }
```
Comment (JessicaLoers) [inline pages/myPlants.js]: 💡 Moving the state from `index.js`  to `_app.js` is a better solution for shared state management across pages, as it allows the plants `data` and the `toggleFavourite `function to be accessible in multiple components without duplicating state logic. 

Please have a look into the Handout from the [global state](https://github.com/neuefische/hh-web-24-8/blob/main/sessions/react-global-state/react-global-state.md) session 

**The advantages:**
- With plants in _app.js, all pages share the same data, ensuring consistency.
- Each page can focus on displaying data without managing its own local storage or duplicate state.
- As the app grows, maintaining centralised state will make it easier to add more features without duplication or synchronization issues.

File: pages/myPlants.js [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+
+  return (
+    <div>
```
Comment (JessicaLoers) [inline pages/myPlants.js]: **As mentioned above.**

- Consider replacing `<div>` with `<main>` here to improve the semantic structure. [The importance of the main element.](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/main) 😁

**Review by JessicaLoers — COMMENTED** [review]
**Great progress on this feature!**  ✨

Here are a few key points for improvement - please also see my comments. 

**State Management:**
- The User Story specifies that users should be able to toggle plants as favorites not only on the list page but also on the Details page. This requires that the `plants state` and `toggleFavourite `function be available to multiple pages.
- To achieve this, it’s essential to lift the plants state to a higher-level component, such as `_app.js`, to make it accessible across different pages and avoid duplicating state logic.

**Semantic HTML:**
- Please consider always using semantic HTML elements like `<main>` instead of `<div>` when wrapping primary content. This helps improve accessibility and SEO and provides clearer structure for assistive technologies.


**Next.js Link Component:**
- When navigating between internal pages, using the `Next.js Link` component instead of `<a>` tags provides optimised client-side navigation, improving the user experience by reducing page load times. 
- So it’s a best practice to use `Link `for all internal routes to make full use of Next.js’s performance benefits. 🎉

**General Layout Component Suggestion:**
- Since the Footer is likely to be used across multiple pages, consider wrapping it in a shared Layout component to centralize structure and styling. This approach simplifies maintenance, ensures a consistent layout, and reduces code duplication.
- For more guidance, you can refer to the Next.js documentation on [Layouts](https://nextjs.org/docs/pages/building-your-application/routing/pages-and-layouts#layout-pattern).

File: components/Footer.js (hunk 3 above)
Comment (Roland-Hufnagel) [reply components/Footer.js]: Why didn't you change this?

File: components/Footer.js [hunk 11]
```diff
@@ -0,0 +1,33 @@
...
+  background-color: lightgray;
+  padding: 10px;
+  display: flex;
+  justify-content: space-around;
+`;
+
+const StyledLink = styled(Link)`
+  text-decoration: none;
+  color: black;
+  flex-direction: column;
```
Comment (Roland-Hufnagel) [inline components/Footer.js]: since this is not a flex-box this will effect nothing.

File: pages/index.js [hunk 12]
```diff
@@ -1,5 +1,17 @@
 import PlantList from "@/components/PlantList";
+import Footer from "@/components/Footer";
+import styled from "styled-components";
 
-export default function Homepage() {
-  return <PlantList />;
+export default function Homepage({ plants, toggleFavourite }) {
+  return (
+    <main>
```
Comment (Roland-Hufnagel) [inline pages/index.js]: you have the `<main>` tag already in your Layout Component so please remove it here.

File: components/PlantList.js [hunk 13]
```diff
@@ -1,29 +1,22 @@
...
 import PlantCard from "./PlantCard";
 
-export default function PlantList() {
+export default function PlantList({ plants, toggleFavourite }) {
   if (!plants || plants.length === 0) return <p>No plants found</p>;
 
   return (
-    <div>
-      <StyledHeader>My Plant Collection</StyledHeader>
+    <main>
```
Comment (Roland-Hufnagel) [inline components/PlantList.js]: here's another unneccessary `<main>` tag.

File: components/PlantCard.js [hunk 14]
```diff
@@ -1,22 +1,30 @@
...
     </Container>
   );
 }
 
-const StyledLink = styled.a`
+const StyledLink = styled(Link)`
```
Comment (Roland-Hufnagel) [inline components/PlantCard.js:33]: Have a look at your RoundImage styled component: You should always make use of the Next Image Component instead of an `` tag.

File: pages/myPlants.js [hunk 15]
```diff
@@ -0,0 +1,34 @@
...
+    ? plants.filter((plant) => plant.isFavourite)
+    : [];
+
+  return (
+    <main>
+      <StyledHeader>My Plants</StyledHeader>
+      {favouritePlants.length > 0 ? (
+        <PlantList plants={favouritePlants} toggleFavourite={toggleFavourite} />
+      ) : (
+        <>
```
Comment (Roland-Hufnagel) [inline pages/myPlants.js]: You don't need fragments here.

File: pages/myPlants.js [hunk 16]
```diff
@@ -0,0 +1,34 @@
...
+          <StyledParagraph>
+            No favourite plants found. Please select some favourite plants.
+          </StyledParagraph>
+        </>
+      )}
+      <Footer />
```
Comment (Roland-Hufnagel) [inline pages/myPlants.js]: Remove the `<Footer/>` component from here. You have already implemented it in the Layout component.

File: package.json [hunk 17]
```diff
@@ -15,7 +15,9 @@
     "next": "13.4.8",
     "react": "18.2.0",
     "react-dom": "18.2.0",
-    "styled-components": "^6.0.2"
+    "remixicon": "^4.5.0",
```
Comment (Roland-Hufnagel) [inline package.json:18]: What is remixicon and where do you use it?

File: pages/[id].js [hunk 18]
```diff
@@ -68,7 +71,7 @@ export default function PlantDetails() {
   );
 }
 
-const Container = styled.div`
+const Container = styled.main`
```
Comment (Roland-Hufnagel) [inline pages/[id].js]: Here's another <main> Element. You have a lot of mains nested in other mains. Your site should always have only one <main> Element.

File: pages/[id].js [hunk 19]
```diff
@@ -36,30 +36,33 @@ export default function PlantDetails() {
...
+        <p aria-label="Fertiliser season">
           Fertiliser season:
           {plant.fertiliserSeason.map((season) => (
-            <span key={season}>
-              {seasonIcons[season] || "🌱"} {season}{" "}
+            <span key={season} aria-label={`Season: ${season}`}>
+              {seasonIcons[season] || "🌱"} {season}
             </span>
           ))}
         </p>
```
Comment (Roland-Hufnagel) [inline pages/[id].js]: I recommend to create a PlantDetails component and render it here. You can copy and paste this code to the component. The reason is separating of concerns. This file (`[id].js`) is for rendering the page e.g. with some standard layout but for rendering the deails info it's a good practice to have a reusable component.

**Review by Roland-Hufnagel — CHANGES_REQUESTED** [review]
Hey team. You already did a great job 💪. To keep a good and clean code right from the beginning I left some comments I want you to look at.

File: package.json (hunk 17 above)
Comment (IboNdure) [reply package.json:18]: we use it for our icons

**Review by Roland-Hufnagel — APPROVED** [review]
Looks great now 💪🦈
